src/invoice/errors.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt::Display;
+use std::num::ParseIntError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub struct VerificationError(pub molecule::error::VerificationError);
+
+impl PartialEq for VerificationError {
+    fn eq(&self, _other: &Self) -> bool {
+        false
+    }
+}
+impl Eq for VerificationError {}
+impl Display for VerificationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+#[derive(Error, PartialEq, Debug)]
+pub enum InvoiceError {
+    #[error("Bech32 error: {0}")]
+    Bech32Error(bech32::Error),
+    #[error("Molecule error: {0}")]
+    MoleculeError(VerificationError),
+    #[error("Failed to parse amount: {0}")]
+    ParseAmountError(ParseIntError),
+    #[error("Unknown prefix")]
+    BadPrefix,
+    #[error("Unknown currency")]
+    UnknownCurrency,
```

Comment:
Include also the currency string so that user may spot out misspelling immediately.